chore(docs): Update Contributing section, RFC process, Governance Model (#34900)
gatsbyjs/gatsby: docs/contributing/gatsby-governance-model.md

Gatsby is an [open source](https://opensource.org/) project supported by [Gatsby, Inc.](https://www.gatsbyjs.com/about/) employees and an active community of contributors.

> Our objectives for this document are to provide current information about the management and maintenance of Gatsby open source software (OSS) and present a variety of resources and avenues of communication between Gatsby’s leadership and community. It also includes a detailed description of contributor levels, code review process, pull request merge process and moderation. Furthermore, it outlines how you can get involved with Gatsby OSS. Establishing this governance models will help support active and productive communication/collaboration within the Gatsby community and between the community and the company.

Reviewer: ```suggestion
Our objectives for this document are to provide current information about the management and maintenance of Gatsby open source software (OSS) and present a variety of resources and avenues of communication between Gatsby’s leadership and community. It also includes a detailed description of contributor levels, accepted contribution types, the voting process, and moderation. Furthermore, it outlines how you can get involved with Gatsby OSS. Establishing this governance models will help support active and productive communication/collaboration within the Gatsby community and between the community and the company.
```

**All community members must follow the [Code of Conduct (CoC)](/contributing/code-of-conduct/).** Consequences for CoC violations are detailed in [moderation](#moderation).

## Get involved

All contributions are welcome, including issues, contributing code, new docs as well as updates and tweaks, blog posts, helping out people, and more. Read the [How to Contribute guide](/contributing/#how-to-contribute) to learn what the community can do for you and what you can do for the community.

### Gatsby User Collective

The [Gatsby User Collective](https://gatsbyuc.dev/)'s goal is to democratize the maintenance of community plugins to allow more folks to assist in their maintenance and to create higher quality plugins in the Gatsby ecosystem. Contributing to the Gatsby User Collective is a valuable contribution!

### Areas for contribution

> As the core of Gatsby is a complex piece of software, it requires some onboarding before contributions can be made efficiently. We're clearly defining here in which areas we accept contributions at the moment and in which not. This is especially important for new features as they always have an inherent maintenance cost -- and making changes without the complete context around Gatsby are difficult here. _This list is not set in stone and can change when e.g. a "Core Maintainer (L3)" role is introduced._

Reviewer: ```suggestion
As the core of Gatsby is a complex piece of software, it requires some onboarding before contributions can be made efficiently. We're clearly defining here in which areas we accept contributions at the moment and in which not. This is especially important for new features as they always have an inherent maintenance cost -- and making changes without the complete context around Gatsby is difficult. _This list is not set in stone and can change when e.g. a "Core Maintainer (L3)" role is introduced._
```

Also if Team is actually L3 (I'm not sure if it is) then "Core Maintainer (L3)" should be "Core Maintainer (L4)"

> We accept any contribution to:

Reviewer: ```suggestion
We accept all types of contributions to:
```

- [`examples`](https://github.com/gatsbyjs/gatsby/tree/master/examples)
- [`starters`](https://github.com/gatsbyjs/gatsby/tree/master/starters)
- [`integration-tests`](https://github.com/gatsbyjs/gatsby/tree/master/integration-tests)
- [`e2e-tests`](https://github.com/gatsbyjs/gatsby/tree/master/e2e-tests)
- [`docs`](https://github.com/gatsbyjs/gatsby/tree/master/docs)

> We accept any contribution to [`packages`](https://github.com/gatsbyjs/gatsby/tree/master/packages) except for these items:

Reviewer: ```suggestion
We accept all types of contributions to [`packages`](https://github.com/gatsbyjs/gatsby/tree/master/packages) with these exceptions:
```

- [`gatsby`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby)
- [`gatsby-core-utils/remote-file-utils`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-core-utils/src/remote-file-utils)

> For the items above we only accept contributions like bug fixes, documentation updates etc. If you want to implement a new feature you have to open a [feature request](#feature-requests) first.

Reviewer: ```suggestion
For the areas above we accept a limited set of contribution types, such as bug fixes and documentation updates. To implement new features in these areas you must open a [feature request](#feature-requests) first.
```

- Invitation to the `@Maintainer` role on [Discord](https://gatsby.dev/discord)
- Invitation to the `Maintainers` team on GitHub.
- New name color on Discord: **Blue**.
- Invitation to the private `#maintainers` channel on Discord.
- Ability to moderate Discord to remove spam, harmful speech, etc.
- Ability to join the `@Moderator` role on Discord (optional, opt-in).
- Ability to review GitHub PRs.
- Ability to vote on _some_ initiatives (see [Voting](#voting) below).

#### Responsibilities

- Participate in the project as a team player.
- Bring a friendly, welcoming voice to the Gatsby community.
- Be active on Discord, especially in the "Help" category.
- Triage new issues.
- Review pull requests.
- Merge some, non-trivial community pull requests.
- Merge your own pull requests (once reviewed and approved).

#### Nomination

To be nominated, a nominee is expected to already be performing some of the responsibilities of a Maintainer over the course of at least a couple of weeks.

In some rare cases, this role may be revoked by an **Admin**. However, under normal circumstances this role is granted for as long as the contributor wishes to engage with the project.

> ### Team

Reviewer: Team is referenced as L3 on line 180, if it is L3 it would be good to have that in the heading. Since it didn't look like the other level headings I got confused for a bit and didn't know we were talking about a new level

If you’d like to learn more about how we approach the relationship between our commercial and open source work, check out this blog post: [Founding Organizations: Creating a Company That Sustains Our Open-Source Community](/blog/2020-02-11-founding-organizations/).

### Feature requests

Please open a [feature request on GitHub Discussions](https://github.com/gatsbyjs/gatsby/discussions/categories/ideas-feature-requests) to voice your ideas. The **Team** will review incoming requests and give a first assessment.

### RFCs

Please see the dedicated [RFC Process document](/contributing/rfc-process/) for more details.

## Voting

Certain project decisions (like membership nominations) require a vote. Below are the changes that require a vote, and the rules that govern that vote.

> One **Admin** may initiate a vote for any unlisted project decision. This person then will be the leading Admin for this vote and handle all necessary actions. General rules will apply, along with any addition rules provided at the admin's discretion. If this unlisted project decision is expected to be repeated in the future, voting rules should be agreed on and then added to this document.

Reviewer: ```suggestion
One **Admin** may initiate a vote for any unlisted project decision. This person then will be the leading Admin for this vote and handle all necessary actions. General rules will apply, along with any additional rules provided at the admin's discretion. If this unlisted project decision is expected to be repeated in the future, voting rules should be agreed on and then added to this document.
```
